commercetools.Sdk/commercetools.Base.Client/Tokens/AnonymousSessionTokenProvider.cs

```diff
@@ -10,19 +10,19 @@ public class AnonymousSessionTokenProvider : TokenProvider, ITokenProvider
             IClientConfiguration configuration,
             HttpClient httpClient,
             IAnonymousCredentialsStoreManager anonymousCredentialsStoreManager,
-            ITokenSerializerService serializerService)
-            : base(httpClient, anonymousCredentialsStoreManager, serializerService)
+            ITokenSerializerService serializerService,
+            string tokenEndpointBaseAddress)
+            : base(httpClient, anonymousCredentialsStoreManager, serializerService, tokenEndpointBaseAddress)
         {
             this.ClientConfiguration = configuration;
             this.anonymousCredentialsStoreManager = anonymousCredentialsStoreManager;
         }
 
         public TokenFlow TokenFlow => TokenFlow.AnonymousSession;
-
-        public override HttpRequestMessage GetRequestMessage()
+        
+        protected override string BuildTokenRequestUri()
         {
-            HttpRequestMessage request = new HttpRequestMessage();
-            string requestUri = this.ClientConfiguration.AuthorizationBaseAddress + $"oauth/{this.ClientConfiguration.ProjectKey}/anonymous/token?grant_type=client_credentials";
+            var requestUri = this.TokenEndpointBaseAddress+"?grant_type=client_credentials";
```

Comment:
all the query parameters should be moved to the body and not be added to the URL. Could you please double check this also for the other TokenProviders?

And it would be good to set a `Content-Type: application/x-www-form-urlencoded` header too